include/SDL3/SDL_clipboard.h

```diff
@@ -36,6 +36,34 @@
 extern "C" {
 #endif
 
+/**
+ * \brief Clipboard content container
+ *
+ * This container struct is used to prepare and hold multiple mime-types of
+ * clipboard content to provide to the OS such as but not limited to images.
+ *
+ * The structure is a linked list allowing multiple mime-types to be offered up
+ * as clipboard data at the same time.
+ *
+ * The internal values within the struct should not need to be manipulated
+ * manually instead refer to the defined functions relating to this struct.
+ *
+ * \since This struct is available since SDL 3.0.0.
+ *
+ * \sa SDL_CreateClipboardData
+ * \sa SDL_AddClipboardData
+ * \sa SDL_SetClipboardData
+ * \sa SDL_GetClipboardData
+ * \sa SDL_HasClipboardData
+ */
+typedef struct SDL_ClipboardData
+{
+    void *data;                         /**< A pointer to the clipboard data */
```

Comment:
As was brought up in the related issue. Data could be a callback function instead allowing for data to be generated when needed. Personally I'm not a fan but I'm open to change to that method if that is what is generally preferred.

Reply:
> In my "goal usecase" which made me create the issue I had "taking screenshots" in mind. Which will not benefit from the on demand approach.

Given the example in the PR description:
> Eg. Offer an image both as jpeg and png

the on-demand approach would be useful, as you would have to generate only the requested format, without wasting time and memory generating the other(s).

Reply:
You still need to create the SDL_Surface from the SDL_Renderer (SDL_RenderReadPixels) which is a heavy operation. The generation from SDL_Surface to image types is very minor in comparison. The only benefit I see then is memory at the cost of a more complex implementation where the "source" SDL_Surface needs to be kept around.